### PR TITLE
FEAT-#2629: implement usecols for read_csv with OmniSci backend

### LINKED ISSUE #2629: Implement usecols parameter for read_csv with OmniSci backend
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. What kind of performance improvements would you like to see with this new API?
According to pyarrow documentation `pyarrow.read_csv` supports `include_columns` (https://arrow.apache.org/docs/python/generated/pyarrow.csv.ConvertOptions.html#pyarrow.csv.ConvertOptions), it can be used for implementation of `usecols` parameter of `modin.read_csv` with OmniSci backend.

### PR DESCRIPTION
**Pull request by amyskov**
Signed-off-by: Alexander Myskov <alexander.myskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2629  
- [x] tests added and passing

### CODE REVIEW

**Review by ienkovich — COMMENTED** [review]
The patch looks good! Have few minor comments.

**Review by ienkovich — APPROVED** [review]

File: modin/experimental/engines/omnisci_on_ray/io.py [hunk 1]
```diff
@@ -229,3 +234,55 @@ def _dtype_to_arrow(cls, dtype):
...
+                usecols_md = sorted(usecols_md)
+                if len(column_names) < max(usecols_md):
+                    raise NotImplementedError(
+                        "max usecols value is higher than the number of columns"
+                    )
+                usecols_md = [column_names[i] for i in usecols_md]
+            elif callable(usecols_md):
+                usecols_md = [
+                    col_name for col_name in column_names if usecols_md(col_name)
+                ]
```
[the remark this answers is not in the document]
Comment (amyskov) [reply modin/experimental/engines/omnisci_on_ray/io.py:286]: Fixed.

File: modin/experimental/engines/omnisci_on_ray/io.py [hunk 2]
```diff
@@ -229,3 +234,55 @@ def _dtype_to_arrow(cls, dtype):
...
+                        col_name for col_name in column_names if col_name in usecols_md
+                    ]
+                else:
+                    raise NotImplementedError(
+                        "values passed in the `usecols` parameter don't match columns names"
+                    )
+            elif usecols_names_dtypes == "integer":
+                # columns should be sorted because pandas doesn't preserve columns order
+                usecols_md = sorted(usecols_md)
+                if len(column_names) < max(usecols_md):
```
[the remark this answers is not in the document]
Comment (amyskov) [reply modin/experimental/engines/omnisci_on_ray/io.py]: Fixed.

File: modin/experimental/engines/omnisci_on_ray/io.py [hunk 3]
```diff
@@ -229,3 +234,55 @@ def _dtype_to_arrow(cls, dtype):
...
+    def _define_usecols(cls, read_csv_kwargs):
+        """
+        Define `usecols` parameter in the way pyarrow can process it.
+        ----------
+        read_csv_kwargs:
+                read_csv function parameters.
+
+        Returns
+        -------
+        usecols_md:
```
[the remark this answers is not in the document]
Comment (amyskov) [reply modin/experimental/engines/omnisci_on_ray/io.py]: Value type was added.

File: modin/experimental/engines/omnisci_on_ray/io.py [hunk 4]
```diff
@@ -229,3 +234,55 @@ def _dtype_to_arrow(cls, dtype):
             return pa.string()
         else:
             return pa.from_numpy_dtype(tname)
+
+    @classmethod
+    def _define_usecols(cls, read_csv_kwargs):
```
[the remark this answers is not in the document]
Comment (amyskov) [reply modin/experimental/engines/omnisci_on_ray/io.py]: Corrected.

**Pull request merged**